starcheck/calc_ccd_temps.py

```diff
@@ -290,32 +333,33 @@ def calc_model(model_spec, states, start, stop, aacccdpt=None, aacccdpt_times=No
     return model
 
 
-def get_week_states(tstart, tstop, bs_cmds, tlm):
+def get_week_states(rltt, sched_stop, bs_cmds, tlm):
     """
-    Make states from last available telemetry through the end of the backstop commands
+    Make states from last available telemetry through the end of the schedule
 
-    :param tstart: start time from first backstop command
-    :param tstop: stop time from last backstop command
+    :param rltt: running load termination time (discard running load commands after rltt)
+    :param sched_stop: create states out through scheduled stop time
     :param bs_cmds: backstop commands for products under review
     :param tlm: available pitch and aacccdpt telemetry recarray from fetch
     :returns: numpy recarray of states
     """
     # Get temperature data at the end of available telemetry
-    ok = tlm['time'] > tlm['time'][-1] - 1400
-    init_aacccdpt = np.mean(tlm['aacccdpt'][ok])
-    init_tlm_time = np.mean(tlm['time'][ok])
+    times = tlm['time']
+    i0 = np.searchsorted(times, times[-1] - 1400)
+    init_aacccdpt = np.mean(tlm['aacccdpt'][i0:])
+    init_tlm_time = np.mean(tlm['time'][i0:])
 
-    # Get commands from last telemetry up to (but not including)
-    # first backstop command.
-    cmds = kadi.commands.get_cmds(init_tlm_time, tstart)
+    # Get currently running (or approved) commands from last telemetry up to
+    # and including commands at RLTT
+    cmds = kadi.commands.get_cmds(init_tlm_time, rltt, inclusive_stop=True)
 
     # Add in the backstop commands
     cmds = cmds.add_cmds(bs_cmds)
 
     # Get the states for available commands.  This automatically gets continuity.
     state_keys = ['obsid', 'pitch', 'q1', 'q2', 'q3', 'q4', 'eclipse']
-    states = kadi_states.get_states(cmds=cmds, state_keys=state_keys,
-                                    merge_identical=True)
+    states = kadi_states.get_states(cmds=cmds, start=init_tlm_time, stop=sched_stop,
```

Comment:
I had been thinking of the get_states changes as a convenience.  Here, now that `start` is passed, it looks like the first returned state in the test load I'm viewing (JAN117A), now starts 15 minutes earlier at `init_tlm_time` instead of the time of `cmds[0]`.  I *think* that is totally fine for starcheck's thermal model but I wasn't expecting the change.

Reply:
More than fine, it is now correct.  Previously the propagation temperature from available telemetry was being applied starting at the wrong time. In most circumstances it didn't really matter, but it is theoretically possible that the first command could be many hours after `init_tlm_time`, e.g. during a long observation between comms.

Reply:
That was my thinking as well (that it is more correct), though I figured it was worth calling out in discussion in the PR as it is a functional change that was not mentioned in the PR top matter.